wrapper/pre_copy.py

```diff
@@ -0,0 +1,682 @@
+        STATE.disks = self.disks
+        STATE.write()
+
+    def _fix_disks(self, domxml):
+        class DiskToFix(object):
+            __slots__ = ['path', 'fixed']
+
+            def __init__(self, path):
+                self.path = path
+                self.fixed = False
+
+        disk_map = {disk.path: DiskToFix(disk.local_path)
+                    for disk in self.disks}
+        tree = ETree.fromstring(domxml)
+        for disk in tree.find('devices').findall('disk'):
+            src = disk.find('source')
+            if src is None:
+                continue
+            path = src.get('file')
+            if path is None:
+                continue
+            disk_data = disk_map.get(path)
+            if disk_data is None:
+                continue
+            # disk.set('type', 'block')
+            # del src.attrib['file']
+            # src.set('dev', dm['path'])
+            driver = ETree.Element('driver')
+            driver.set('type', 'raw')
+            src.set('file', disk_data.path)
+            disk.append(driver)
+            disk_data.fixed = True
+
+        # Check that all paths were changed
+        for k, v in six.iteritems(disk_map):
+            if not v.fixed:
+                raise RuntimeError('Disk path "%s" was '
+                                   'not fixed in the domxml' % k)
+
+        return ETree.tostring(tree)
+
+    def get_xml(self):
+        xmlfile = os.path.join(self._tmp_dir_path, 'vm.xml')
+        with open(xmlfile, 'wb') as f:
+            f.write(self._fix_disks(self.vmware.get_domxml()))
+        return xmlfile
+
+    def _get_nbdkit_cmd(self, disk, vmware_password_file, filters):
+        env = 'LD_LIBRARY_PATH=%s' % VDDK_LIBRARY_PATH
+        if 'LD_LIBRARY_PATH' in os.environ:
+            env += ':' + os.environ['LD_LIBRARY_PATH']
+
+        nbdkit_cmd = [
+            'env',
+            env,
+            'nbdkit',
+            '-v',
+            '-U', disk.sock,
+            '-P', disk.pidfile,
+            '--exit-with-parent',
+            '--readonly',
+            '--foreground',
+            '--exportname=/',
+        ] + [
+            '--filter=' + f for f in filters
+        ] + [
+            '--filter=log',
+            'vddk',
+            # pylint: disable=protected-access
+            'vm=moref=%s' % self.vmware.get_vm()._moId,
+            'server=%s' % self.vmware.server,
+            'password=+%s' % vmware_password_file,
+            'thumbprint=%s' % self.vmware.thumbprint,
+            'libdir=%s' % VDDK_LIBDIR,
+            'file=%s' % disk.path,
+        ]
+        if self.vmware.user:
+            nbdkit_cmd.append('user=%s' % self.vmware.user)
+        nbdkit_cmd.extend([
+            'logfile=%s' % STATE.wrapper_log,
+            'logappend=true',
+        ])
+
+        return nbdkit_cmd
+
+    def _start_nbdkits(self, vmware_password_file):
+        paths = []
+        filters = ['cacheextents', 'retry', 'blahaha']
```

Comment:
That is a leftover from testing. Did you really have to remove it in order for it to work? I tried it with this so that I am sure it gets removed and then forgot to remove it before submitting the PR. I already noticed it ;)

Reply:
Yes, I really had to remove it. It seems that the loop stops after the first exception, so only the first non existing filter is removed. On a RHEL 8 machine, both `retry` and `'blahaha` are absent, so `blahaha` is not removed and the conversion fails.